[Gen 2] Add Guest Access documentation (#7227)
aws-amplify/docs: src/pages/[platform]/build-a-backend/auth/guest-access/index.mdx

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';

export const meta = {
  title: 'Enable guest access',
  description: 'Access services without needing to sign in.',
  platforms: [
    'javascript',
    'react-native',
    'flutter',
    'swift',
    'android',
    'angular',
    'nextjs',
    'react',
    'vue'
  ]
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta
    }
  };
}

> <InlineFilter filters={['flutter','swift','android']}>

Reviewer: it looks like these two inline filters are the same? are they needed?

Author: the URLs are different for these platforms.

The AWS Cognito Auth Plugin can be configured to automatically obtain guest credentials once the device is online so that you are able to use other categories "anonymously" without the need to sign in. You will not be able to perform user specific methods while in this state such as updating attributes, changing your password, or getting the current user. However, you can obtain the unique Identity ID which is assigned to the device through the `fetchAuthSession` method [described here](/[platform]/build-a-backend/auth/accessing-credentials/).

> Amplify Gen 2 enables guest access by default. To disable it, you can update the `backend.ts` file with the following changes:

Reviewer 2: ```suggestion
```

Author: I think we don't need this because of the change below, right? @josefaidt

Reviewer 2: the text will be duplicated with https://github.com/aws-amplify/docs/pull/7227/files#diff-cdd72c3d4a00ba2a2685935330f96609322050f491b0cb69160437b2f3c050a8R43

</InlineFilter>
<InlineFilter filters={['javascript','react-native','angular','nextjs','react','vue']}>

> The AWS Cognito Auth Plugin can be configured to automatically obtain guest credentials once the device is online so that you are able to use other categories "anonymously" without the need to sign in. You will not be able to perform user specific methods while in this state such as updating attributes, changing your password, or getting the current user. However, you can obtain the unique Identity ID which is assigned to the device through the `fetchAuthSession` method [described here](/[platform]/build-a-backend/auth/manage-user-session/).

Reviewer 2: ```suggestion
Amplify can be configured to automatically obtain guest credentials once the device is online so that you are able to use other categories "anonymously" without the need to sign in. You will not be able to perform user specific methods while in this state such as updating attributes, changing your password, or getting the current user. However, you can obtain the unique Identity ID which is assigned to the device through the `fetchAuthSession` method [described here](/[platform]/build-a-backend/auth/manage-user-session/).
```

> Amplify Gen 2 enables guest access by default. To disable it, you can update the `backend.ts` file with the following changes:
> 
> </InlineFilter>

Reviewer 2: ```suggestion
</InlineFilter>

Amplify Gen 2 enables guest access by default. To disable it, you can update the `backend.ts` file with the following changes:
```

```ts
const backend = defineBackend({
  auth,
  data,
});

const { cfnIdentityPool } = backend.auth.resources.cfnResources;
cfnIdentityPool.allowUnauthenticatedIdentities = false;
```

> 

Reviewer 3: @josefaidt is this actually needed? I thought we do this automatically now?

Author: I asked about this to Josef and he mentioned it is enabled by default. This is for disabling.
